stdlib/ballerina-database/src/main/java/org/ballerinalang/database/sql/SQLDataIterator.java

```diff
@@ -130,11 +135,19 @@ public BStruct generateNext() {
                     SQLColumnDefinition def = (SQLColumnDefinition) columnDef;
                     columnName = def.getName();
                     int sqlType = def.getSqlType();
+                    BStructType.StructField[] structFields = this.type.getStructFields();
                     ++index;
+                    int fieldTypeTag = this.type.getStructFields()[index - 1].getFieldType().getTag();
+                    boolean isOriginalValueNull = false;
                     switch (sqlType) {
                     case Types.ARRAY:
                         Array dataArray = rs.getArray(index);
-                        bStruct.setRefField(++refRegIndex, getDataArray(dataArray));
+                        if (fieldTypeTag == TypeTags.UNION_TAG) {
+                            handleUnionTypeStructField(bStruct, structFields, index, ++refRegIndex,
+                                    getDataArray(dataArray), TypeTags.ARRAY_TAG);
+                        } else {
+                            bStruct.setRefField(++refRegIndex, getDataArray(dataArray));
+                        }
```

Comment:
If all the values in the column are null in DB, then it will return an array of null values. So shouldn't we handle that as well?